File: app/src/main/java/me/gilo/starter/dao/DaoAccess.java
package me.gilo.starter.dao;


import androidx.lifecycle.LiveData;
import androidx.room.*;
import me.gilo.starter.models.Note;

import java.util.List;


@Dao
public interface DaoAccess {

    @Insert
    Long insertTask(Note note);


    @Query("SELECT * FROM Note ORDER BY created_at desc")
    LiveData<List<Note>> fetchAllTasks();


    @Query("SELECT * FROM Note WHERE id =:taskId")
    LiveData<Note> getTask(int taskId);


    @Update
    void updateTask(Note note);


    @Delete
    void deleteTask(Note note);
}

File: localdataprovider/src/main/java/me/gilo/localdataprovider/dao/DaoAccess.kt
package me.gilo.localdataprovider.dao


import androidx.lifecycle.LiveData
import androidx.room.*
import me.gilo.localdataprovider.models.NoteEntity


@Dao
interface DaoAccess {

    @Insert
    fun insertTask(noteEntity: NoteEntity): Long?


    @Query("SELECT * FROM NoteEntity ORDER BY created_at desc")
    fun fetchAllTasks(): LiveData<List<NoteEntity>>


    @Query("SELECT * FROM NoteEntity WHERE id =:taskId")
    fun getTask(taskId: Int): LiveData<NoteEntity>


    @Update
    fun updateTask(noteEntity: NoteEntity)


    @Delete
    fun deleteTask(noteEntity: NoteEntity)
}
